keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1462,225 +1278,309 @@ prevented. Deleting the entire cluster also leaves resources allocated outside
-    // If null (or empty), it is assumed this allocation will be processed by a
-    // single kubelet plugin with no ResourceHandle data attached. The name of
-    // the kubelet plugin invoked will match the DriverName set in the
-    // ResourceClaimStatus this AllocationResult is embedded in.
+    // A driver may allocate devices provided by other drivers, so this
+    // driver name here can be different from the driver names listed in
+    // the results.
+    //
+    // This is an alpha field and requires enabling the DRAControlPlaneController
+    // feature gate.
+    //
+    // +optional
+    ControllerName string
+
+    // This field holds configuration for all drivers which
+    // satisfied requests in this claim. The configuration applies to
+    // the entire claim.
+    //
+    // +optional
+    // +listType=atomic
+    Config []ConfigurationParameters
+
+    // Results lists all allocated devices.
     //
     // +listType=atomic
-    ResourceHandles []ResourceHandle
+    Results []RequestAllocationResult
 
-    // This field will get set by the resource driver after it has allocated
-    // the resource to inform the scheduler where it can schedule Pods using
-    // the ResourceClaim.
+    // Setting this field is optional. If unset, the allocated devices are available everywhere.
     //
-    // Setting this field is optional. If null, the resource is available
-    // everywhere.
     // +optional
-    AvailableOnNodes *core.NodeSelector
+    AvailableOnNodes *v1.NodeSelector
 
-    // Shareable determines whether the resource supports more
-    // than one consumer at a time.
+    // Shareable determines whether the claim supports more than one consumer at a time.
+    //
     // +optional
     Shareable bool
 }
 
-// AllocationResultResourceHandlesMaxSize represents the maximum number of
-// entries in allocation.resourceHandles.
-const AllocationResultResourceHandlesMaxSize = 32
+// AllocationResultsMaxSize represents the maximum number of
+// entries in allocation.results.
+const AllocationResultsMaxSize = 32
 
-// ResourceHandle holds opaque resource data for processing by a specific kubelet plugin.
-type ResourceHandle struct {
-    // DriverName specifies the name of the resource driver whose kubelet
-    // plugin should be invoked to process this ResourceHandle's data once it
-    // lands on a node. This may differ from the DriverName set in
-    // ResourceClaimStatus this ResourceHandle is embedded in.
+// RequestAllocationResult contains configuration and the allocation result for
+// one request.
+type RequestAllocationResult struct {
+    // DriverName specifies the name of the DRA driver whose kubelet
+    // plugin should be invoked to process the allocation once the claim is
+    // needed on a node.
+    //
+    // Must be a DNS subdomain and should end with a DNS domain owned by the
+    // vendor of the driver.
     DriverName string
```

Comment:
In light of changing to `ControllerName` should this change to `PluginName`?

Reply:
Then we would also have to renamed it in the ResourcePoolSlice, which then makes it awkward when that object gets published by a driver (sic!) which isn't a kubelet plugin.

I prefer to think of it as "a DRA driver has a name and provides a kubelet plugin". The plugin registers under the name of the driver.

Reply:
I know that the kubelet code uses "plugin name". That confused me each time I read that code.